dask_ml/decomposition/pca.py

```diff
@@ -431,18 +472,25 @@ def inverse_transform(self, X):
         If whitening is enabled, inverse_transform does not compute the
         exact inverse operation of transform.
         """
-        check_is_fitted(self, "mean_")
+        check_is_fitted(self, "components_")
+
+        if self.center:
+            check_is_fitted(self, "mean_")
```

Comment:
It seems that we always set `self.mean_`. Is this necessary in this if block?

Reply:
Yes, indeed - `self.mean_` gets always set.
My implementation choice has been for the check to be inside the `if` block here, for the purpose of demonstrating intent (i.e., we utilize the `mean_` parameter here, only if we're performing the centering step).
The `mean_` parameter is still computed at `fit` time, regardless.

Reply:
If you meant, whether it's _at all_ necessary to check if `mean_` has been set - yes, probably. Because this is in `transform()`, and - as it can be seen from the diff - the same check existed beforehand, it was just moved to indicate logically in what circumstance we need to utilize the `mean_` value.